plugins/scaffolder-backend/src/scaffolder/stages/publish/github.ts

```diff
@@ -28,8 +31,15 @@ export class GithubPublisher implements PublisherBase {
     config: GitHubIntegrationConfig,
     { repoVisibility }: { repoVisibility: RepoVisibilityOptions },
   ) {
+    config.accessType = 'oAuth';
```

Comment:
Why do we need to know about the access type here? I'm wondering if the CredentialsProvider should return the type there instead. So then we can use the `token` returned from the CredentialsProvider always rather than looking at the config at all.

Reply:
We need to know this because of the way the auth should be written to. 
If it is an oauth token: 
```
auth = {
      username: this.config.token,
      password: 'x-oauth-basic',
};
``` 
But if it is an githubApp: 
```
auth = {
      username: 'x-access-token',
      password: this.config.token,
};
```

Reply:
We could also initialize the `credentialsProvider` as undefined and only initialize it if the config.token does not exists. And then in the publisher we could use it being defined to alter the way the auth is created.

Reply:
Something like:
```
...
let credentialsProvider: GithubCredentialsProvider|undefined = undefined;

    if (!config.token) {
      if (!config.apps) {
        return undefined;
      }
      credentialsProvider = GithubCredentialsProvider.create(config);
    }
...
```

Reply:
I did a new commit using the `credentialsProvider` to determine the type of auth being used.

Reply:
Made it a separate change because it was affecting more packages, but once https://github.com/backstage/backstage/pull/4541 is in it simplifies things by not having to know what kind of token we're dealing with.

Reply:
@Rugvip if you check the latest commit I reverted the unnecessary change to the github config

Reply:
Oh... I see what you mean now... I misunderstood